docs/meta.rst

```diff
@@ -11,24 +11,38 @@ Taichi provides metaprogramming infrastructures. Metaprogramming can
 
 Taichi kernels are *lazily instantiated* and a lot of computation can happen at *compile-time*. Every kernel in Taichi is a template kernel, even if it has no template arguments.
 
+
+.. _template_metaprogramming:
+
+Template metaprogramming
+------------------------
+
+.. code-block:: python
+
+    @ti.kernel
+    def copy(x: ti.template(), y: ti.template()):
+        for i in x:
+            y[i] = x[i]
```

Comment:
Lack of description, add a `TODO: WIP` mark or complete in this PR.

Reply:
Most part of meta.rst is still WIP, so I wouldn't mark every section as WIP. I'll, of course, take a pass there later.

Reply:
In fact, `meta` is already translated in docs-zh-cn...

Reply:
That's fine. I'll only add some content there.